### PR TITLE
New Table API Template

### PR DESCRIPTION
**Pull request by markjbrown**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-cosmosdb-table/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+				"failoverPriority": 1
+			}
+		]
+	},
+	"resources": 
+	[
+		{
+			"type": "Microsoft.DocumentDB/databaseAccounts",
+			"name": "[variables('accountName')]",
+			"apiVersion": "2015-04-08",
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply 101-cosmosdb-table/azuredeploy.json]: update it to what? This API matches our resource provider version

File: 101-cosmosdb-table/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "accountName": {
+      "value": "GEN-UNIQUE"
+    },
+    "location": {
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply 101-cosmosdb-table/azuredeploy.parameters.json]: done

File: 101-cosmosdb-table/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+				"locations": "[variables('locations')]",
+				"databaseAccountOfferType": "Standard",
+				"enableAutomaticFailover": "[parameters('automaticFailover')]",
+				"enableMultipleWriteLocations": "[parameters('multipleWriteLocations')]"
+			}
+		},
+		{
+			"type": "Microsoft.DocumentDB/databaseAccounts/apis/tables",
+			"name": "[concat(variables('accountName'), '/table/', parameters('tableName'))]",
+			"apiVersion": "2015-04-08",
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply 101-cosmosdb-table/azuredeploy.json]: I don't know what this should be. The date here is correct for our RP

File: 101-cosmosdb-table/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,157 @@
+{
+	"$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"accountName": {
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply 101-cosmosdb-table/azuredeploy.json]: done

File: 101-cosmosdb-table/azuredeploy.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmoore-msft) [reply 101-cosmosdb-table/azuredeploy.json]: your manifest shows 2 later versions, use the latest unless you have a reason not to:   "2016-03-31"

[the remark this answers is not in the document]
Comment (BunnyShanZhang) [reply 101-cosmosdb-table/azuredeploy.json]: @bmoore-msft  Why can I only see the version 2015-04-08 on this site https://docs.microsoft.com/en-us/azure/templates/microsoft.documentdb/2015-04-08/databaseaccounts/apis/databases/containers? Where can I find the related document of 2016-03-31. Seems that there are some differences between these two versions.

**Pull request merged**